internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -192,6 +197,36 @@ func (e *EnvStackConfig) Parameters() ([]*cloudformation.Parameter, error) {
 	return e.transformParameters(currParams, e.prevParams, transformEnvControllerParameters)
 }
 
+func getSecurityGroupConfig(mft *manifest.Environment) (*template.SecurityGroupConfig, error) {
+	if mft != nil && !mft.Network.VPC.SecurityGroupConfig.IsEmpty() {
```

Comment:
if you have a large set of nested-ifs that's usually a hint at using guard clauses: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html